### PR TITLE
feat(mapless_architecture): add mapless_architecture

### PR DESCRIPTION
**Pull request by simon-eisenmann-driveblocks**
## Description

We created a mission planner with locally-generated, sensor-based maps as described in https://github.com/orgs/autowarefoundation/discussions/4545. The entire project is contained within the mapless_architecture folder, which should be added to the repository.

## Related links

- GitHub discussion: https://github.com/orgs/autowarefoundation/discussions/4545
- Related PR in the autoware_msgs repository: https://github.com/autowarefoundation/autoware_msgs/pull/95

## How was this PR tested?

This PR was tested using rosbags and unit tests.

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp
Comment (maxime-clem) [inline planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp:1]: `main` files like this one should be removed. Please see how nodes are implement in newer Autoware packages (e.g., `autoware_motion_velocity_planner_node`).

File: planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp [hunk 1]
```diff
@@ -0,0 +1,196 @@
+// Copyright 2024 driveblocks GmbH
+// driveblocks proprietary license
+#ifndef AUTOWARE__MISSION_LANE_CONVERTER__MISSION_LANE_CONVERTER_NODE_HPP_
+#define AUTOWARE__MISSION_LANE_CONVERTER__MISSION_LANE_CONVERTER_NODE_HPP_
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "autoware_auto_planning_msgs/msg/path.hpp"
+#include "autoware_auto_planning_msgs/msg/trajectory.hpp"
```
Comment (maxime-clem) [inline planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp:9]: Messages should be from `autoware_planning_msgs`.

File: planning/mapless_architecture/autoware_mission_lane_converter/Readme.md
Comment (maxime-clem) [inline planning/mapless_architecture/autoware_mission_lane_converter/Readme.md:1]: Please add the basic information about the node in this README (see other nodes' README files). There should be a table with the interface (input/output topics) and the node parameters.

File: planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+# install launchfile(s)/param file(s)
+install(DIRECTORY
+  launch
+  param
+  DESTINATION share/${PROJECT_NAME})
+
+# --- SPECIFY TESTS ---
+# configure clang format
+set(ament_cmake_clang_format_CONFIG_FILE ../../.clang-format)
```
Comment (maxime-clem) [inline planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt:37]: this should not be necessary

File: planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  find_package(ament_lint_auto REQUIRED)
+
+  ament_auto_add_gtest(${PROJECT_NAME}_tests test/gtest_main.cpp
+    test/src/test_mission_planner_converter.cpp
+    src/mission_lane_converter_node.cpp)
+
+  ament_lint_auto_find_test_dependencies()
+endif()
+
+ament_auto_package()
```
Comment (maxime-clem) [inline planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt:50]: `ament_auto_...` are not necessary. Please check other `CMakeLists.txt` files (e.g., in the `autoware_path_smoother`).

**Review by maxime-clem — CHANGES_REQUESTED** [review]

File: planning/.pages [hunk 4]
```diff
@@ -40,6 +40,7 @@ nav:
       - 'Algorithm': planning/autoware_freespace_planning_algorithms
       - 'RRT*': planning/autoware_freespace_planning_algorithms/rrtstar
   - 'Mission Planner': planning/autoware_mission_planner
+  - 'Local Mission Planner': planning/mapless_architecture
```
[the remark this answers is not in the document]
Comment (simon-eisenmann-driveblocks) [reply planning/.pages:43]: Done!

File: planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp:1]: Done! Should I place the hpp files in the src folder as in autoware_motion_velocity_planner_node?

File: planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt (hunk 2 above)
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt:37]: I removed it!

File: planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt (hunk 3 above)
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_mission_lane_converter/CMakeLists.txt:50]: Done!

File: planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp (hunk 1 above)
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp:9]: Done!

File: planning/mapless_architecture/autoware_mission_lane_converter/Readme.md
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_mission_lane_converter/Readme.md:1]: Done!

File: planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp (hunk 1 above)
Comment (maxime-clem) [reply planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp:9]: I guess the commit was not pushed :sweat_smile:

File: planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp
Comment (maxime-clem) [reply planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp:1]: As you like.
It is okay to put the `.hpp` together with the `.cpp` files in the `src` folder for packages whose code is not meant to be included in other packages.

File: planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp (hunk 1 above)
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_mission_lane_converter/include/autoware/mission_lane_converter/mission_lane_converter_node.hpp:9]: Yes, I will push all the changes when the internal review is finished!

File: planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp
Comment (simon-eisenmann-driveblocks) [reply planning/mapless_architecture/autoware_local_mission_planner/src/mission_planner_main.cpp:1]: Ok, thanks for the answer! I will keep the `.hpp` files in the include folder!

**Comment by simon-eisenmann-driveblocks** [comment]
I pushed the requested changes!

**Comment by simon-eisenmann-driveblocks** [comment]
PR will be done here: https://github.com/tier4/mapless_autoware/pull/3

**Pull request closed without merging**